commec/setup.py

```diff
@@ -0,0 +1,692 @@
+        self.default_taxonomy_download_url: str = (
+            "ftp://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz"
+        )
+        self.taxonomy_download_url: str = self.default_taxonomy_download_url
+
+        if automate:
+            self.download_biorisk = True
+            self.download_blastnr = True
+            self.download_blastnt = True
+            self.download_taxonomy = True
+            self.do_setup()
+        else:
+            self.start()
+
+    def start(self):
+        """
+        Starts the user interrogation process.
+        """
+        print(
+            """\n                       Welcome to\n
+ ██████╗ ██████╗ ███╗   ███╗███╗   ███╗███████╗ ██████╗ \033[38;5;202m         ▄▄               \033[0m
+██╔════╝██╔═══██╗████╗ ████║████╗ ████║██╔════╝██╔════╝ \033[38;5;202m       ▄███▌              \033[0m
+██║     ██║   ██║██╔████╔██║██╔████╔██║█████╗  ██║      \033[38;5;202m      ▐█████              \033[0m
+██║     ██║   ██║██║╚██╔╝██║██║╚██╔╝██║██╔══╝  ██║      \033[38;5;202m     ▐██████▌             \033[0m
+╚██████╗╚██████╔╝██║ ╚═╝ ██║██║ ╚═╝ ██║███████╗╚██████╗ \033[38;5;202m     ███████▌             \033[0m
+ ╚═════╝ ╚═════╝ ╚═╝     ╚═╝╚═╝     ╚═╝╚══════╝ ╚═════╝ \033[38;5;202m    ▐███████▌             \033[0m
+\033[48;5;17m█ █████▄ █████▄ █ ▄█▀█▄                                 \033[38;5;202m     ███████   ▄█▄      \033[0m
+\033[48;5;17m█ █    █ █    █ █ █   ▀          DATABASE               \033[38;5;202m      █████▌  ▄███▄▄     \033[0m  
+\033[48;5;17m█ █████▄ █████▄ █ ▀███▄            SETUP                \033[38;5;202m      ▐█████▄██▀    ▀▄    \033[0m   
+\033[48;5;17m█ █    █ █    █ █ ▄   █              UTILITY            \033[38;5;202m      ▐████████       ▌  \033[0m
+\033[48;5;17m█ █████▀ █████▀ █ ▀█▄█▀                                 \033[38;5;202m      ████████▀         \033[0m
+                                                        \033[38;5;202m   ▄▄██████▀▀             \033[0m
+                                                        \033[38;5;202m ▀▀                       \033[0m"""
+        )
+        print(
+            "                 The Common Mechanism!",
+            "\n\nInternational Biosecurity and Biosafety Initiative for Science",
+            "\nCopyright © 2021-2024 ",
+            "\n\nThis script will help download the mandatory databases ",
+            "\nrequired for using Commec Screen, and requires a stable",
+            "\ninternet connection, wget, and update_blastdb.pl.",
+        )
+        self.print_help_info()
+        print()
+        self.check_requirements()
+        self.setup_overall_directory()
+        self.do_setup()
+
+    def print_help_info(self, additional_help=[""]):
+        """
+        Prints helpful instructions according to the current step,
+        as well as general instructions.
+        """
+        add_help_str = "" if len(additional_help) == 0 else "".join(additional_help)
+        print(
+            "\033[38;5;202m"
```

Comment:
Changing the colours is VERY persistant, haha, so if someone happens to ctrl+c in the ms time-frame when the wget etc are being checked, then yes, they are left with a grey text colour, because it was never unset.
This can be handled by as you point out, ensuring every individual print statement is responsible for resetting the colours, however that can lead to many more repeats.

Factoring out the ANSI hack codes into constants works well, but requires string concatenation considerations. Which can be clunky in some cases. I was actually thinking about doing this before the last set of commits, but thought heck, it works, haha I'll go back and give them constants for readibilities sake.